docroot/modules/custom/mass_alerts/tests/src/ExistingSiteJavascript/AlertsPlacementTest.php

```diff
@@ -5,13 +5,13 @@
 use Behat\Mink\Exception\ExpectationException;
 use Drupal\mass_content_moderation\MassModeration;
 use Drupal\paragraphs\Entity\Paragraph;
-use weitzman\DrupalTestTraits\ExistingSiteWebDriverTestBase;
+use weitzman\DrupalTestTraits\ExistingSiteSelenium2DriverTestBase;
 use weitzman\LoginTrait\LoginTrait;
 
 /**
  * Test Alerts Placement.
  */
-class AlertsPlacementTest extends ExistingSiteWebDriverTestBase {
+class AlertsPlacementTest extends ExistingSiteSelenium2DriverTestBase {
```

Comment:
Fixing this both explained so much and left me confused.

`chromedriver` is suppose to speak the webdriver protocol. `selenium` is also the webdriver protocol. So why do we need separate base classes? Does chromedriver speak webdriver, but also extend it with additional non-webdriver APIs like the call to `http://chromedriver:9515/json/version`?

Reply:
I'm actually not that clear on this part of DTT either. Last we spoke @jibran wanted to remove `ExistingSiteWebDriverTestBase` and just keep ExistingSiteSelenium2DriverTestBase  so perhaps we dont need to puzzle over this.

Reply:
I think I figured it out, but totally unintentionally. I wanted to see why a test wasn't failing on develop, and I wanted to see the browser live. `chromedriver` wasn't working, throwing 400 responses. After inspecting the existing container, it's using _headless chrome_ with the [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/). At a basic level, that means running Chrome directly with `/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --remote-debugging-address=0.0.0.0 --remote-debugging-port=9222` (macOS firewall for me blocks the port externally, and I was lazy so I just set `0.0.0.0`).

I think the confusion comes from various naming:

- `\DMore\ChromeDriver\ChromeDriver` is older than the `chromedriver` project. The former class uses the DevTools protocol, while `chromedriver` speaks WebDriver.
- `\weitzman\DrupalTestTraits\WebDriverTrait` and `\weitzman\DrupalTestTraits\ExistingSiteWebDriverTestBase` both use `\DMore\ChromeDriver\ChromeDriver`, so they use the DevTools protocol.

If this all makes sense, I think we should file a few pull requests updating documentation and class headers to make this all clearer.

Reply:
It doesnt really make sense to me, but thats not your fault. I'm not really looking to become a Chrome expert. I would be more than happy to merge those PRs :)

Reply:
Filed https://gitlab.com/weitzman/drupal-test-traits/-/merge_requests/109